client/post-editor/media-modal/index.jsx

```diff
@@ -174,6 +180,11 @@ export const EditorMediaModal = React.createClass( {
 
 		MediaActions.delete( site.ID, toDelete );
 		analytics.mc.bumpStat( 'editor_media_actions', 'delete_media' );
+		const mediaIds = map( toDelete, 'ID' );
+		this.props.deleteMedia( site.ID, mediaIds );
+		if ( includes( mediaIds, siteIconId ) ) {
+			this.props.resetSiteIcon( site.ID );
+		}
```

Comment:
I'm a bit concerned about this code, what if we allow media deletion from elsewhere, we'll face the exact same problem. Do we have the media URLs available here, that way we could use them to compare against the siteIcon attribute in the site reducer? 

Or we could solve this differently, move this "extra" behavior to the deleteMedia action creator. You could call to requestSiteSettings action creator from there. I know it's not perfect, but maybe this could allow us to "centralize" this behavior, which seem not directly related to this component? Thoughts? I may be missing something tho :)